pinot-core/src/main/java/com/linkedin/pinot/core/io/readerwriter/RealtimeIndexOffHeapMemoryManager.java

```diff
@@ -63,6 +81,7 @@ public PinotDataBuffer allocate(long size, String columnName) {
     PinotDataBuffer buffer = allocateInternal(size, columnName);
     _totalMemBytes += size;
     _buffers.add(buffer);
+    _serverMetrics.addValueToTableGauge(_tableName, ServerGauge.REALTIME_OFFHEAP_MEMORY_USED, size);
```

Comment:
For gauge, we should report total value at this instance which is totalMemBytes. size variable only represents incremental change.

Reply:
we will get a per-table value at the current level. We are adding value to the table guage on each alloc. Alternatively, we can set total MBytes, yes. The result is the same.